src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -521,6 +550,31 @@ def makeTransferRucio(self, wflow, dataIn, dids, dataSize, grouping, copies, nod
             self.logger.info(msg, wflow.getName(), dids, rseExpr, ruleAttrs)
         return success, transferId
 
+    def getRids(self, workflowName):
+        """
+        Obtain transfer IDs for given workflow name
+        :param workflowName: workflow name
+        :return: list of transfer IDs
+        """
+        # make request to MSTransferor service
+        # https://xxx.cern.ch/ms-transferor/data/info?request=<workflowName>
+        tids = []
+        mgr = RequestHandler()
```

Comment:
Instead of using pycurl, we could reuse the code already available in this module to retrieve a workflow document similarly to how it is done [here](https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/MicroService/MSMonitor/MSMonitor.py#L44C1-L44C69). I guess it would be something like `transferRecord = self.reqmgrAux.getTransferInfo(<your_workflow_name>)`

In different words, the same document can be retrieved either from MSTransferor or from ReqMgr2. Example:
https://cmsweb.cern.ch/ms-transferor/data/info?request=cmsunified_HARVEST-Run2022E_ZeroBias_30May2023v2_240123_140316_7935
and
https://cmsweb.cern.ch/reqmgr2/data/transferinfo/cmsunified_HARVEST-Run2022E_ZeroBias_30May2023v2_240123_140316_7935

Reply:
Alan, you contradict yourself, see https://github.com/dmwm/WMCore/pull/12155#discussion_r1852228457 , where you explicitly ask to use ms-transferor and now you are asking to use ReqMgr2 (which by the way I originally implemented). Please make your mind which service should be used. My understanding from your original comment was that since we changing MSTransferor code it can be used to get this info too.

Reply:
Valentin, I didn't notice that the same document can be retrieved from two different sources.

And the comment you pointed out does not have anything to do with this discussion. You had implemented something completely different, which was basically asking Rucio for rules for a given workflow. Rucio does not know any information about our workflows.

Please use ReqMgrAux, which is already fully integrated into this microservice (in practice, it has the same behavior as ms-transferor, but with less code changes and dependencies.

Reply:
ok, done